examples/triangle.rs

```diff
@@ -11,20 +11,34 @@ fn main() {
     // Initialize wgpu
     let event_loop: EventLoop<()> = EventLoop::new();
     let window = winit::window::Window::new(&event_loop).unwrap();
-    let instance = wgpu::Instance::new(wgpu::Backends::PRIMARY);
-    let surface = unsafe { instance.create_surface(&window) };
+    let instance = wgpu::Instance::new(wgpu::InstanceDescriptor {
+        backends: wgpu::Backends::PRIMARY,
+        dx12_shader_compiler: wgpu::Dx12Compiler::default(),
+    });
+    let surface = unsafe {
+        instance
+            .create_surface(&window)
+            .expect("Failed to create wgpu surface")
+    };
     let adapter =
         futures::executor::block_on(instance.request_adapter(&Default::default())).unwrap();
     let (device, queue) =
         futures::executor::block_on(adapter.request_device(&Default::default(), None)).unwrap();
-    let swapchain_format = surface.get_supported_formats(&adapter)[0];
+    let caps = surface.get_capabilities(&adapter);
+    let formats = caps.formats;
+    let swapchain_format = *formats.get(0).expect("No supported formats for surface");
+    let mut view_formats = vec![];
+    if swapchain_format.describe().srgb == false {
+        view_formats.push(swapchain_format.add_srgb_suffix());
+    }
```

Comment:
```suggestion
    let swapchain_format = surface.get_capabilities(&adapter).formats[0];
```

Reply:
The view_formats stuff is so that it looks correct on webgpu (apparently webgpu doesn't support sRGB swapchains? according to what I've been told on the wgpu matrix), and works around a bug in wgpu 0.15 (that's getting fixed in 0.15.1) where sRGB formats aren't getting sorted to the front of the list of supported formats.

Reply:
These code snippets aren't really meant as a "getting start with wgpu" so much as a minimal boilerplate to have a runnable example. Correctly configuring sRGB feels out of scope, unless the example truly won't run in certain environments